pkg/client/selfhosted/selfhosted.go

```diff
@@ -245,9 +248,13 @@ func (c *Client) doRequest(ctx context.Context, url, header string, obj interfac
 	req = req.WithContext(ctx)
 	if len(c.Bearer) > 0 {
 		req.Header.Add("Authorization", "Bearer "+c.Bearer)
+	} else {
+		req.SetBasicAuth(c.Username, c.Password)
```

Comment:
I'm wondering if we should check the existence of anything in username/password here? Incase someone is using a fully public endpoint/registry